basics-rest-spring/controller/src/main/java/com/epam/esm/controller/CertificateController.java

```diff
@@ -0,0 +1,85 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.service.Impl.CertificateService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/certificate")
+public class CertificateController {
+
+    private final CertificateService service;
+
+    public CertificateController(CertificateService service) {
+        this.service = service;
+    }
+
+    @GetMapping(path = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Certificate> findCertificate(@PathVariable Integer id) {
+        return new ResponseEntity<>(service.find(id), HttpStatus.OK);
+    }
+
+    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<Certificate>> findAll() {
+        return new ResponseEntity<>(service.findAll(), HttpStatus.OK);
+    }
+
+    @GetMapping(path = "/byDate", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<Certificate>> sortedCertificatesByDate() {
```

Comment:
Try to find examples how to add sorting.  Example
site.com/books?filter=author%3DGeorge%20Orwell%26year%3C%3D1945&order_by=price